apps/src/applab/manageAssets.jsx

```diff
@@ -0,0 +1,138 @@
+var React = require('react');
+var AssetsApi = require('./manageAssets/clientApi');
+var AssetRow = require('./manageAssets/assetRow.jsx');
+
+var errorMessages = {
+  415: 'This type of file is not supported.',
+  500: 'The server responded with an error.'
+};
+
+function getErrorMessage(status) {
+  return errorMessages[status] || 'An unknown error occurred.';
+}
+
+module.exports = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func,
+    typeFilter: React.PropTypes.string
+  },
+
+  getInitialState: function () {
+    return {
+      assets: null,
+      uploadStatus: ''
+    };
+  },
+
+  componentWillMount: function () {
+    // TODO: Use Dave's client api when it's finished.
+    AssetsApi.ajax('GET', '', function (xhr) {
```

Comment:
Thinking more on this, it's the hidden access to the dashboard.project.current.id global in AssetsApi that I don't like. If ClientApi were a class that could be instantiated with a project id, or if the ajax call received the project id as a parameter, then I'd be happier. In either case the project id (or whatever object or method wraps it) should be passed into the component as a prop.

I know this sounds annoying, but I think a big benefit of React is being really deliberate about where state (and props) are stored and passed through the app.